web_monitoring/cli.py

```diff
@@ -0,0 +1,55 @@
+# Command Line Interface
+# All the functionality is implemented in this module. See scripts/ for
+# executables.
+from docopt import docopt
+import hashlib
+import io
+import lxml.html
+import requests
+from tqdm import tqdm
+from web_monitoring import internetarchive as ia
+from web_monitoring import db
+
+
+# These functions lump together library code into monolithic operations for the
+# CLI. They also print. To access this functionality programmatically, it is
+# better to use the underlying library code.
+
+
+def import_ia(url, agency, site):
+    print('obtaining versions list from Internet Archive...')
+    versions = ia.list_versions(url)
+    # Collect all the results and POST them as a unit (rather than streaming).
+    formatted_versions = []
+    for dt, uri in tqdm(versions, desc='formatting versions'):
+        res = requests.get(uri)
+        assert res.ok
+        version_hash = hashlib.sha256(res.content)
+        content_as_file = io.StringIO(res.content.decode(errors='ignore'))
```

Comment:
pardon my not taking time to understand the purpose of the `content_as_file` var, but you'd probably want to avoid any sort of coercion during string conversion. the `decode` func looks suspicious to me, as it's clearly trying to intepret the underlying response bytes, which aren't necessarily UTF-8 (character encoding is supposed to be specified setting the Content-Type header's charset property). Any sort of coercion here runs the risk of beaking the hash on line 27. If I'm understanding the purpose of `content_as_file` correctly, you'd just want to do a straight string representation of the underlying byte array.